lib/kino/smart_cell/server.ex

```diff
@@ -0,0 +1,118 @@
+defmodule Kino.SmartCell.Server do
```

Comment:
This has a number of similarities to `Kino.JS.Live.Server` and we can easily extract some pieces, but for now I kept it separate to make further changes easier.